agents-common/src/main/java/org/apache/ranger/plugin/model/validation/RangerPolicyValidator.java

```diff
@@ -958,6 +958,22 @@ private String getDuplicate(List<String> values) {
 		}
 		return duplicate;
 	}
+	private void removeDuplicates(List<String> values){
+		if (values==null || values.isEmpty()){
+			return;
+		}
+		Iterator<String> itr = values.iterator();
+		HashSet<String> uniqueElements = new HashSet<>();
```

Comment:
Consider replacing #965 to #975 with the following:

```
Set<String> uniqueElements = new HashSet<>();

for (Iterator<String> iter = values.iterator(); iter.hasNext(); ) {
  if (!uniqueElements.add(iter.next())) {
    iter.remove();
  }
}

```

Reply:
Consider this version as well:

```
Set<String> uniqueElements = new HashSet<>();

values.removeIf(e -> !uniqueElements.add(e));
```

Reply:
Fixed in latest patch